hazenlib/tasks/acr_uniformity.py

```diff
@@ -78,73 +78,53 @@ def get_integral_uniformity(self, dcm):
-            img, self.ACR_obj.dx, self.ACR_obj.dy
-        )
+        mask=self.ACR_obj.get_mask_image(img)
+        masked_img=mask * img
+        cx, cy = mask.shape[1] // 2, mask.shape[0] // 2
+        (centre_x, centre_y) = (cx,cy)
+
         # Dummy circular mask at centroid
-        base_mask = ACRObject.circular_mask(
-            (centre_x, centre_y + d_void), r_small, dims
-        )
+        base_mask = ACRObject.circular_mask((centre_x, centre_y + d_void), r_small, dims)
         coords = np.nonzero(base_mask)  # Coordinates of mask
 
         # TODO: ensure that shifting the sampling circle centre
         # is in the correct direction by a correct factor
         lroi = self.ACR_obj.circular_mask([centre_x, centre_y + d_void], r_large, dims)
         img_masked = lroi * img
-        half_max = np.percentile(img_masked[np.nonzero(img_masked)], 50)
-
-        min_image = img_masked * (img_masked < half_max)
-        max_image = img_masked * (img_masked > half_max)
-
-        min_rows, min_cols = np.nonzero(min_image)[0], np.nonzero(min_image)[1]
-        max_rows, max_cols = np.nonzero(max_image)[0], np.nonzero(max_image)[1]
+        rows, cols = (img)[0], (img)[1]
 
         mean_array = np.zeros(img_masked.shape)
 
-        def uniformity_iterator(masked_image, sample_mask, rows, cols):
-            """Iterates spatially through the pixel array with a circular ROI and calculates the mean non-zero pixel
-            value within the circular ROI at each iteration.
-
-            Args:
-                masked_image (np.array): subset of pixel array.
-                sample_mask (np.array): _description_.
-                rows (np.array): 1D array.
-                cols (np.array): 1D array.
-
-            Returns:
-                np.array: array of mean values.
-            """
-            # Coordinates of mask
-            coords = np.nonzero(sample_mask)
-            for idx, (row, col) in enumerate(zip(rows, cols)):
-                centre = [row, col]
-                translate_mask = [
-                    coords[0] + centre[0] - centre_x - d_void,
-                    coords[1] + centre[1] - centre_y,
-                ]
-                values = masked_image[translate_mask[0], translate_mask[1]]
-                if np.count_nonzero(values) < np.count_nonzero(sample_mask):
-                    mean_val = 0
-                else:
-                    mean_val = np.mean(values[np.nonzero(values)])
-
-                mean_array[row, col] = mean_val
-
-            return mean_array
-
-        min_data = uniformity_iterator(min_image, base_mask, min_rows, min_cols)
-        max_data = uniformity_iterator(max_image, base_mask, max_rows, max_cols)
-
-        sig_max = np.max(max_data)
-        sig_min = np.min(min_data[np.nonzero(min_data)])
-
-        max_loc = np.where(max_data == sig_max)
-        min_loc = np.where(min_data == sig_min)
-
-        piu = 100 * (1 - (sig_max - sig_min) / (sig_max + sig_min))
+        results = []
```

Comment:
please provide more description of what sort of results this list is storing
either as a comment in the code or by updating the variable name to something more descriptive

Reply:
I've now added comments throughout my code changes